src/core/FactoryMaker.js

```diff
@@ -85,23 +80,21 @@ let FactoryMaker = (function () {
 
     function getSingletonFactory(classConstructor) {
         return function (context) {
-
+            let instance;
             if (context === undefined) {
                 context = {};
             }
-
-            let instance = getSingletonInstance(context, classConstructor.__dashjs_factory_name);
-
             return {
                 getInstance: function () {
-
                     if (instance) {
                         return instance;
+                    } else {
```

Comment:
Ah yes since we do this singletonContext mapping we do not need traditional instance check for singleton ...   Will clean that up today.  thanks for the note.